crates/lsp/src/lib.rs

```diff
@@ -249,7 +249,26 @@ impl<L: LSPLang> Backend<L> {
   async fn publish_diagnostics(&self, uri: Url, versioned: &VersionedAst<StrDoc<L>>) -> Option<()> {
     let mut diagnostics = vec![];
     let path = uri.to_file_path().ok()?;
-    let rules = self.rules.for_path(&path);
+
+    let rules = match &self.rules {
```

Comment:
This will make every file reports wrong diagnostics. Would it be better to report error only once?